src/qibo/quantum_info/metrics.py

```diff
@@ -29,10 +33,12 @@ def purity(state):
     else:
         pur = np.real(np.trace(np.dot(state, state)))
 
+    pur = float(pur)
```

Comment:
I see that this is done in many functions, why is it needed?

Reply:
Otherwise it returns a 0-dim array with a float inside.